api/prisma/schema.prisma

```diff
@@ -7,40 +7,204 @@ datasource db {
   url      = env("MONGOHQ_URL")
 }
 
-type DonationStartDate {
-  date DateTime @map("_date") @db.Date
-  when String   @map("_when")
+// USER COLLECTION ---------------------
+
+type FileData {
+  contents String
+  ext      String
+  key      String
+  name     String
+  path     String?
+
+  // Nonesense
+  dataSource Json?    @map("__dataSource") // Null only | Undefined
+  persisted  Boolean? @map("__persisted") // Undefined
+  strict     Boolean  @map("__strict") // Undefined
 }
 
-type UserBadges {
-  /// Could not determine type: the field only had null or empty values in the sample set.
-  coreTeam Json?
+type File {
+  contents String
+  ext      String
+  key      String
+  name     String
+  path     String? // Undefined | Null
+
+  // Nonesense
+  cachedRelations Json?     @map("__cachedRelations") // Undefined
+  data            FileData? @map("__data") // Undefined
+  dataSource      Json?     @map("__dataSource") // Null | Undefined
+  persisted       Boolean?  @map("__persisted") // Undefined
+  strict          Boolean?  @map("__strict") // Undefined
 }
 
-type UserCompletedChallenges {
+type CompletedChallengeData {
+  challengeType Int? // Null | Undefined
   completedDate Float
   id            String
+  solution      String? // Null | Undefined
+
+  // Nonesense
+  dataSource Json?   @map("__dataSource") // Null
+  persisted  Boolean @map("__persisted")
+  strict     Boolean @map("__strict")
 }
 
-type UserProfileUi {
+type CompletedChallenge {
+  challengeType      String? // Null | Undefined
+  completedDate      Float
+  files              File[]
+  githubLink         String? // Undefined
+  id                 String
+  isManuallyApproved Boolean? // Undefined
+  solution           String? // Undefined
+
+  // Nonsense
+  cachedRelations Json?                   @map("__cachedRelations")
+  data            CompletedChallengeData?
```

Comment:
```suggestion
  data            CompletedChallengeData? @map("__data") // Undefined
```
Since we have no plans to use the Nonsense, can we mark everything with `__` prefixing it at optional?

Reply:
Not if it is always there... Its parent might not always be there, but provided `a`, then `a.b` always exists.